packages/modules/web-viewer-react/src/components/Viewer.tsx

```diff
@@ -3,13 +3,25 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 
+import { IModelApp } from "@itwin/core-frontend";
 import { BaseViewer } from "@itwin/viewer-react";
-import React from "react";
+import React, { useEffect } from "react";
 
 import { useWebViewerInitializer } from "../hooks";
 import type { WebViewerProps } from "../types";
 
 export const Viewer = (props: WebViewerProps) => {
   const initialized = useWebViewerInitializer(props);
+  useEffect(() => {
+    const handleBeforeUnload = async () => {
+      IModelApp.shutdown()
```

Comment:
take a look at `WebInitializer.cancel`

also wonder if we need to manually dispose of other stuff ourselves like presentation, but wonder if they listen to an event on imodelapp shutdown

Reply:
Other stuff would have to register a [onbeforeShutdown](https://github.com/iTwin/itwinjs-core/blob/3cc2df33f158c51d02fb17a3c3d4d303c857acd8/core/frontend/src/IModelApp.ts#L442) event, and it seems to imply that they would have to manually be disposed. Is this event handled by us, or the team in charge of that particular stuff, I'm not sure.

If we pass IModelApp.shutdown() into WebInit.cancel(), we would have to be careful of the side-effects of `.shutdown()`. Because inherently, we are looking to free up resources, and a question we would have is 'how many resources do we really need to free up` to prevent this memory leak? We know that `_renderSystem` needs to be disposed, but what about the other things disposed in shutdown(), like the authClient, entityClasses, toolAdmin, viewManager, and tileAdmin?

Reply:
i looked at WebInitializer.cancel and init(), two things popped up to me - IModelApp does have a [extensionAdmin](https://github.com/iTwin/itwinjs-core/blob/74610de97a26fc395842b34adf63dbfd7c9432e4/core/frontend/src/extension/ExtensionAdmin.ts#L44), which the web viewer calls its' registerHost() on init if an extension is registered. IModelApp.shutdown() doesn't do anything dispose wise for extensionAdmin, do you think we need to do anything there?

Only other thing would be [ViewerAuthorization.client](https://github.com/iTwin/viewer/blob/b4d784da8456f8043548c004b6d8506e20cacaf4/packages/modules/web-viewer-react/src/services/Initializer.ts#L59), it will hold a reference to an authClient, that wouldn't be cleared up, but I don't think this is that big of an issue?

Reply:
Theres nothing apart of the extension admin that i think needs to be disposed
It should be ok to just set IModelApp.authorizationClient = undefined, we dont want to dispose of the actual client

Reply:
does appui need to be shut down?

Reply:
no, appui doesn't - the root of the problem lies in any possible webgl textures not being disposed of correctly, and that's not something app ui is responsible for. While app ui does have a [UiAdmin](https://github.com/iTwin/itwinjs-core/blob/ec2d5b0b84d08b2dda6bf3a8765bb2feb64ac777/ui/appui-abstract/src/appui-abstract/UiAdmin.ts#L245) class, it does not contain a shutdown() that other admin classes have.